### PR TITLE
Don't waste our precious rate limit analyzing stale PRs

### PR DESCRIPTION
**Pull request by zackgalbreath**

### CODE REVIEW

**Review by scottwittenburg — COMMENTED** [review]
@zackgalbreath Nice, thanks!  You must have gotten a glimpse of this in the logs, but probably not the whole picture.  At some point, it would be great if we could see how often we post duplicated status to a PR.   I'm guessing it's a lot, and the longer PRs sit untouched, the worse it gets.  So I like this idea a lot :+1: 

I have a question regarding this specific change: We know some pipelines will get stuck, in which case, we'll be coming long to cancel/restart them around 24 hrs after creation.  With this PR, will PR authors then have to push some change to their branch in order to get `pull.updated_at` to move forward and get a new status post from us?  Or are you counting on the spooky action at a distance when someone visits the PR in github, causing the `updated_at` to change when github makes the new merge commit?  On the other hand, It's possible I'm forgetting some important detail here.

File: images/gh-gl-sync/SpackCIBridge.py [hunk 1]
```diff
@@ -127,8 +127,16 @@ def list_github_prs(self):
                 print("Skipping draft PR {0} ({1})".format(pull.number, pull.head.ref))
                 backlogged = "draft"
                 push = False
+
             pr_string = "pr{0}_{1}".format(pull.number, pull.head.ref)
 
+            if push and pull.updated_at < datetime.now() + timedelta(minutes=-1440):
+                # Skip further analysis of this PR if it hasn't been updated in 24 hours.
+                # This helps us avoid wasting our rate limit on PRs with merge conflicts.
```
[the remark this answers is not in the document]
Comment (zackgalbreath) [reply images/gh-gl-sync/SpackCIBridge.py:135]: I reused the same "long" threshold we use to query for the currently running `develop` pipeline here: https://github.com/spack/spack-infrastructure/blob/main/images/gh-gl-sync/SpackCIBridge.py#L47

**Comment by zackgalbreath** [comment]
> I have a question regarding this specific change: We know some pipelines will get stuck, in which case, we'll be coming long to cancel/restart them around 24 hrs after creation. With this PR, will PR authors then have to push some change to their branch in order to get `pull.updated_at` to move forward and get a new status post from us? Or are you counting on the spooky action at a distance when someone visits the PR in github, causing the `updated_at` to change when github makes the new merge commit? On the other hand, It's possible I'm forgetting some important detail here.

This is a good point. I don't think it would be a problem if we increased this "stale PR" threshold to 48 hours or so if you think that's a better idea. Ultimately, the purpose of this PR is to provide a shortcut to avoid wasting our limited number of GitHub API queries on PRs like this one:

https://api.github.com/repos/spack/spack/pulls/10466

It hasn't been updated since January, 2020 but we continue to hit the GitHub API for it multiple times, each time this script runs, because at one point in time it managed to acquire a `merge_commit_sha`.

File: images/gh-gl-sync/SpackCIBridge.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (scottwittenburg) [reply images/gh-gl-sync/SpackCIBridge.py:135]: Ahh right, make sense.  Thanks for reminding me that "long" threshold also has the potential to reduce the effectiveness of our "stuck pipeline" mitigation.  What do you think of keeping the "stuck pipeline" threshold 24 hrs, and making the "long" threshold 36 hrs and the new threshold here at least 36 hrs? :shrug:

**Review by scottwittenburg — APPROVED** [review]

**Pull request merged**